api/routes/collections.js

```diff
@@ -0,0 +1,207 @@
+const express = require('express');
+const router = express.Router();
+const db = require('../db');
+const dbUtils = require('../utils/db.js');
+const apiUtils = require('../utils/api.js');
+
+/* Create a collection */
+router.post('/', (req, res) => {
+  let hostname = req.protocol + '://' + req.headers.host;
+
+  // Validate data received
+  if (!("name" in req.body)) {
+    let err = { "Error": "The request object is missing the name field" };
+    return res.status(400)
+      .set({ "Content-Type": "application/json" })
+      .send(err);
+  }
+
+  // Build query object
+  let query = { 
+    text: 'INSERT INTO "UserCollection"("userID", name, "isPrivate") VALUES($1, $2, $3) RETURNING *', 
+    values: [] 
+  };
+
+  // TODO: Add user ID
+  query.values.push(1);
+  query.values.push(req.body.name);
+  query.values.push(req.body.isPrivate || false);
+
+  // Run query
+  db.client.query(query, (err, result) => {
+    if (err) {
+      return res.status(400).send(err);
+    }
+
+    let collection = apiUtils.formatCollection(result.rows[0], hostname);
+    res.status(201)
+      .set({
+        "Content-Type": "application/json",
+        "Content-Location": collection.url
+      })
+      .send(collection);
+  });
+
+});
+
+/* Get a collection */
+router.get('/:collection_id', (req, res) => {
+  let hostname = req.protocol + '://' + req.headers.host;
+
+  // Build query object
+  let query = {
+    text: 'SELECT "UserCollection".id AS "collectionID", "UserCollection".name AS "collectionName", "UserCollection"."isPrivate", "User".id AS "userID", "User".username, "User"."imgFileName" AS "userImage", "Game".*, ratings."overallRating" FROM "UserCollection" INNER JOIN "User" ON "User".id = "UserCollection"."userID" LEFT JOIN "CollectionGame" ON "UserCollection".id = "CollectionGame"."userColllectionID" LEFT JOIN "Game" ON "CollectionGame"."gameID" = "Game".id LEFT JOIN (SELECT "gameID", AVG("overallRating") AS "overallRating" FROM "Review" GROUP BY "gameID") ratings ON "Game".id = ratings."gameID" WHERE "UserCollection".id = $1',
+    values: [req.params.collection_id]
+  }
+
+  // Run query
+  db.client.query(query, (err, result) => {
+    if (err) {
+      return res.status(400).send(err);
+    } else if (!result.rows.length) {
+      err = { "Error": "No collection with this id exists" };
+      return res.status(404)
+        .set({ "Content-Type": "application/json" })
+        .send(err);
+    }
+
+    let collection = apiUtils.formatCollectionGames(result.rows, hostname);
+      res.status(200)
+        .set({ "Content-Type": "application/json" })
+        .send(collection);
+  });
+
+});
+
+/* Update a collection */
+router.patch('/:collection_id', (req, res) => {
+  let hostname = req.protocol + '://' + req.headers.host;
+
+  let query = { text: '', values: [] };
+  let query_fields = [];
```

Comment:
Can we replace `query_fields` with `queryFields`? Unless it's specific an api, we should try to keep things camelCase to match the javascript conventions.

Reply:
This has been updated in this file (as well as in `api/routes/reviews.js` and `api/routes/games.js`)